unboundid/src/com/unboundid/ldap/sdk/ConnectThread.java

```diff
@@ -170,6 +170,9 @@ public void run()
       {
         try
         {
+          final int soTimeout =
+                  Math.max(0, (int) options.getResponseTimeoutMillis());
```

Comment:
What is the default value that is set up for ResponseTimeOutMillis? Do we set it or is it set up for us?

Reply:
It appears that if no value is set it sets `0` based on the `Math.max()` on line 174.
Being said, does `Math.max()` handle `null` gracefully if the getter doesn't return a value?

Reply:
defaults are set on controller side. Its 3000 ms. 
Ideally, its never going to be 0

Reply:
We could handle this pretty gracefully by using:
```java
final int soTimeout = Integer.getInteger(options.getResponseTimeoutMillis(), 0);
```

If you're concerned about the possibility of the int value being negative them the Math method should be run after.

Reply:
Yes, i think getInteger is better option. I Just realized options will be null here. Let me see if i can pass it somehow.